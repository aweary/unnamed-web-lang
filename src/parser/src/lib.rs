#![warn(clippy::all)]

mod parser;

pub mod test_utils;
pub use crate::parser::*;

#[cfg(test)]
mod tests {
    // use crate::test_utils::parse_statement;
    // use insta::assert_debug_snapshot;

    // fn assert_stmt_snapshot(input: &str) {
    //     assert_debug_snapshot!(parse_statement(input));
    // }

    // #[test]
    // fn expr_stmt() {
    //     assert_stmt_snapshot("500;");
    //     assert_stmt_snapshot("40.02;");
    //     assert_stmt_snapshot("\"Hello\";");
    //     assert_stmt_snapshot("[1, 2, \"3\"];");
    // }
}
