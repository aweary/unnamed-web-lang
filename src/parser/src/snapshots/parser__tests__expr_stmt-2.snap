---
source: src/lib.rs
expression: parse_statement(input)
---
Stmt {
    id: NodeId(
        0,
    ),
    kind: Expr(
        Expr {
            id: NodeId(
                0,
            ),
            kind: Lit(
                Lit {
                    span: Span {
                        start: ByteIndex(0),
                        end: ByteIndex(5),
                    },
                    kind: Number(
                        40.02,
                    ),
                },
            ),
            span: Span {
                start: ByteIndex(0),
                end: ByteIndex(5),
            },
            ty: None,
        },
    ),
    span: Span {
        start: ByteIndex(0),
        end: ByteIndex(5),
    },
    has_semi: false,
}
